### ISSUE TITLE
'NitroModules/PromiseHolder.hpp' file not found in iOS

### ISSUE THREAD

**Issue by ospfranco** [mrousavy/nitro#378]
### How were you trying to build the app?

Tried adding react-native-nitro-sqlite and getting the following build error:

/Users/osp/Developer/example/ios/Pods/Headers/Private/RNNitroSQLite/RNNitroSQLite-Swift-Cxx-Umbrella.hpp:23:10 'NitroModules/PromiseHolder.hpp' file not found

I've tried nitro modules `0.17.0` and `0.18.0` but no luck. I'm building from Xcode directly after doing a `pod install` in both old arch and new arch:

![CleanShot 2024-11-30 at 16 15 53@2x](https://github.com/user-attachments/assets/9783e673-9261-4942-a700-bd13790847af)

![CleanShot 2024-11-30 at 16 19 48@2x](https://github.com/user-attachments/assets/868cbfc7-bdd1-49c8-bd3f-d607887a454c)

I'm using yarn, when switching to npm getting the following error:

![CleanShot 2024-11-30 at 17 14 31@2x](https://github.com/user-attachments/assets/73a5be53-9600-4943-8cb8-edacddd5f092)

### Full build logs

```tsx
⬆️
```

### Project dependencies

```json
"@craftzdog/react-native-buffer": "^6.0.5",
    "chai": "^4.3.7",
    "chance": "^1.1.9",
    "clsx": "^2.0.0",
    "events": "^3.3.0",
    "mocha": "^10.7.3",
    "nativewind": "^2.0.11",
    "react": "18.3.1",
    "react-native": "0.76.1",
    "react-native-http-bridge-refurbished": "1.2.9",
    "react-native-nitro-modules": "^0.17.0",
    "react-native-nitro-sqlite": "^9.0.2",
    "react-native-restart": "^0.0.27",
    "react-native-share": "^11.0.3",
    "reflect-metadata": "^0.1.13",
    "stream-browserify": "^3.0.0",
    "util": "^0.12.5"
```

### Nitro Modules Version

0.18.0

### Target platforms

iOS

### Operating system

MacOS

### Can you build the Nitro Modules Example app?

I didn't try (⚠️ your issue might get ignored & closed if you don't try this)

### Additional information

- [ ] I am using Expo
- [ ] I am using Nitrogen (nitro-codegen)
- [X] I have read and followed the [Troubleshooting Guide](https://nitro.margelo.com/docs/troubleshooting).
- [ ] I created a reproduction PR to reproduce this issue here in the nitro repo. (See [Contributing](https://nitro.margelo.com/docs/contributing) for more information)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/nitro/issues) and found none.

**Comment by ospfranco** [comment]
This might have to do with my example linking node_modules from a root folder. I will close this, not the use case nitro was thought for.

**Comment by mrousavy** [comment]
Ah well whoops this might be a relict from the past.
PromiseHolder doesn't exist anymore, I'll fix this asap.

Thanks for reporting!

**Comment by mrousavy** [comment]
Wait I think I made a breaking change in NitroModules... I either re-add PromiseHolder.hpp or we re-generate the specs for SQLite @chrispader

**Comment by chrispader** [comment]
I updated the peer dependency of `react-native-nitro-sqlite` to <=16.2.0 in the last release. You'll have to use Nitro v16.2 for now, but there is already a PR up for updating it.

**Comment by mrousavy** [comment]
Perfect- so this is fixed in [Nitro SQLite 9.1.0](https://github.com/margelo/react-native-nitro-sqlite/releases/tag/v9.1.0)
